pypeit/core/skysub.py

```diff
@@ -607,18 +604,45 @@ def local_skysub_extract(sciimg, sciivar, tilts, waveimg, global_sky, rn2_img, t
                     color = 'orange'
                 ginga.show_trace(viewer, ch, sobjs[iobj].trace_spat, sobjs[iobj].idx, color=color)
         '''
+
     # If requested display the model fits for this slit
-    if SHOW_RESIDS == True:
+    if SHOW_RESIDS:
+        viewer, ch = ginga.show_image((sciimg - skyimage - objimage) * np.sqrt(modelivar) * thismask)
         # TODO add error checking here to see if ginga exists
-        viewer, ch = ginga.show_image((sciimg - skyimage - objimage) * np.sqrt(modelivar)*thismask)
-        # TODO figure out a way to overplot the pixels that were masked in red like as a scatter plot
+        canvas = viewer.canvas(ch._chname)
+        out1 = canvas.clear()
+        out2 = ch.cut_levels(-5.0, 5.0)
+        out3 = ch.set_color_algorithm('linear')
+        # Overplot the traces
         for spec in sobjs:
-            if spec.HAND_EXTRACT_FLAG == False:
-                color = 'green'
+            if spec.HAND_EXTRACT_FLAG is False:
+                color = 'magenta'
             else:
                 color = 'orange'
             ginga.show_trace(viewer, ch, spec.trace_spat, spec.idx, color=color)
-        # TODO figure out a way to set the cuts of the ginga viewer to go from -5 to 5
+
+        # These are the pixels that were masked by the extraction
+        spec_mask, spat_mask = np.where((outmask == False) & (inmask == True))
```

Comment:
should be ```is``` and not ```=```
for both.  or better,   

```np.where(~outmask & inmask)```

Reply:
You are confused here @profxj.  is and not are logical operators but they are *not* bitwise operators. outmask is False will evaluate to a single boolean and not a numpy boolean array.

Reply:
Your second example is correct though.